app/src/main/java/com/ichi2/apisample/AnkiDroidHelper.java

```diff
@@ -224,23 +224,36 @@ public Long addNote(long modelId, long deckId, Map<String, String> data, Set<Str
         return getApi().addNote(modelId, deckId, result, tags);
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
This seems not so good, can't we avoid this at least in the code (ok in unit tests)?

Reply:
In this case, the warning is caused by the quirk with generics and array creation in Java. There is no way to avoid it as far as I know